### PR TITLE
feat: Add Admin & Event analytics dashboards + backend API (#4, #8, #25, #26, #40, #41, #42, bug fix)

### LINKED ISSUE #4: US.02 - Administrator Analytics
As an **administrator**, I would like to view global statistics on:
- number of events 
- number of tickets issued
- participation trends (number of registered attendees who show up to events).

**User points**: 4

**Acceptance test**: Admin logs into dashboard → sees total events, total tickets, and participation trends chart. All values update from backend data and match database counts.

### LINKED ISSUE #8: US.06 - Event Analytics
As a **User**, I would like to have a dashboard per event with stats like tickets issued, attendance rates, and remaining capacity.

**User points**: 8

**Acceptance test**: Organizer opens event → Analytics tab shows tickets issued, attended, attendance rate %, and remaining capacity. Values refresh correctly and handle edge cases (0 tickets, full event).

### LINKED ISSUE #25: Task #02.01 - Administrator Analytics UI
- Analytics page for administrator where they can view:
  - number of events
  - number of tickets
  - participation trends

### LINKED ISSUE #26: Task #02.02 - Administrator analytics backend
- database query for all events
- functions for:
  - total number of events
  - total number of issued tickets
  - participation trends (attended / issued tickets over time)

**User points**: 5

**Acceptance test**: API endpoints return correct counts and ratios when tested with sample event data. Participation trend endpoint returns expected array of (date, rate) pairs.

### LINKED ISSUE #40: Task #06.01 - Backend: Metrics model
• Decide where each number comes from: ticket capacity (from the event), tickets issued (count of tickets), people who attended (checked in).
• Add a way to return these numbers for one event: tickets issued, attended, attendance rate, capacity, remaining capacity.
• Make sure it handles edge cases like zero tickets or missing event.
• Do a quick check that the math looks right.

**User points**: 3

**Acceptance test**: Backend /analytics/:eventId returns JSON object with valid numeric fields. Attendance rate correctly computed as attended / issued × 100.

### LINKED ISSUE #41: Task #06.02 - Frontend: Event Analytics Dashboard (UI)
• Add a dashboard screen for one event (Analytics tab or page).
• Show three simple tiles: Tickets Issued, Attendance (number and percent), Remaining Capacity.
• Optional: a small visual (bar or donut) to compare issued vs attended vs remaining.
• Include loading states and a friendly empty state like “No tickets issued yet.”

**User points**: 3

**Acceptance test**: Frontend loads dashboard → displays metrics correctly from API. Shows “No tickets issued yet” when empty. Renders donut chart comparing attendance vs. capacity.

### LINKED ISSUE #42: Task #06.03 - Integration, Refresh, and Tests
• Hook page up to backend 
• Add a Refresh button or light auto-refresh during live events.
• Show a clear message if something fails, and keep the last good data on screen.

**User points**: 2

**Acceptance test**: Analytics page refreshes live data without page reload. If backend is unavailable, a clear error message appears while last valid metrics remain visible.

### PR DESCRIPTION
**Pull request by dan-gan**
2 Features: **Admin Analytics** and **Event Analytics** (Organization user) with UI dashboards and backend API support. 

Closed #4 
Closes #8 
Closes #25 
Closes #26 
Closes #40 
Closes #41 
Closes #42 

- Admin Analytics (#4, #25, #26): dashboard with total events, tickets, participation trends; backend queries implemented.

- Event Analytics (#8, #40, #41, #42): event-level metrics (tickets issued, attendance, capacity) with visual charts; backend integration & refresh.

- Bug Fix: login null reference when entering admin credentials in organization login; account type validation added.

### CODE REVIEW

File: frontend/src/components/Discover/EventAnalytics.jsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+            setLoading(false);
+            setRefreshing(false);
+        }
+    };
+
+    useEffect(() => {
+        if (eventId) {
+            fetchAnalytics();
+        }
+    }, [eventId, token]);
```
Comment (Alphabot54) [inline frontend/src/components/Discover/EventAnalytics.jsx]: missing fetchAnalytics as dependency to useEffect, this causes warning on running app

**Review by Alphabot54 — APPROVED** [review]
Thanks for the fixes! Looks good!
